File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.js
function solution(array) {
    let answer = 0;
    let res = new Array(1000);
    res.fill(0);

    for (let i = 0; i < array.length; i++) {
        res[array[i]] += 1;
    }

    let max = Math.max(...res);
    let cnt = 0;
    for (let i = 0; i < res.length; i++) {
        if (max === res[i]) {
            answer = i;
            cnt++;
        }
        if (cnt >= 2) {
            return -1;
        }
    }
    return answer;
}  

File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.py
def solution(array):
    if not array:
        return -1 
    max_possible_val = max(array) + 1 if array else 1
    res = [0] * max_possible_val
    for num in array:
        res[num] += 1
    max_freq = 0
    if res:
        max_freq = max(res)
    answer = -1
    count_modes = 0
    for i in range(len(res)):
        if res[i] == max_freq:
            answer = i
            count_modes += 1
        if count_modes >= 2:
            return -1 
    return answer